### ISSUE TITLE
Events displayed 1 hour early/late on the day of daylight saving clock shift

### ISSUE THREAD

**Issue by niklaka** [themouette/jquery-week-calendar#12]
Events are being displayed 1 hour early/late (depends on the directon of clock shift) on the day that clocks are being shifted backward/forward.

**Comment by pverrone** [comment]
Yeah, this has been an ongoing issue for awhile now. There was a fix that was posted this past spring, but it doesn't seem to hold water now. :(

I will post that fix later today when I get a chance. In case it is different than what was rolled into the latest code...

**Comment by pverrone** [comment]
I took another look at the "fix" and it really doesn't do anything at all for the hour shift. It was for calculating the first day of week, which get screwy during the change to and from DST. 

Currently, any repeating appts I have that run from one change to another get wacky. 

This is an urgent issue for me, so I will be working on it. If I come up with a fix, I will post it...

**Comment by pverrone** [comment]
ps... There must be some way to ignore or adjust the offset when we need to. Like, before the OS system switches over to DST. The events only display incorrectly when looking at non DST events while the system is still in DST.

**Comment by rstein2000** [comment]
This is an important issue for me, as well.  I agree the issue stems from an over-dependence on the OS system clock.  I've tried to fix this by including some server-side logic when storing/retrieving events, but the results have been mixed.  In addition, the server-side logic needs to be aware of the time zone the user is working in to generate the correct offset, so it's a complicated solution.

To me, I think the solution will require rewriting the recurring event procedure so that instead of adding (for example) 86,400 seconds to a daily recurring event's start day and time, the procedure utilizes a Javascript/jQuery datetime function to compare a recurring event's time string to the original event's string to see if they both read as (for example) 9:00 a.m.

**Comment by pverrone** [comment]
I've gotten around this by making all times local. It's not a solution that would work for everyone, ie. Responsible programmers, but it works in my situation. :P

**Comment by rstein2000** [comment]
Actually, that's not an unreasonable solution.  I was doing that myself for a while.  And you'd still be able to aggregate general usage data if you needed to as long as you were able to record what timezone a user is in.

I had found a solution using PHP's new datetime objects, but my server is running on CentOS, and wouldn't you know, there is a bug involving PHP datetime objects when run on a CentOS server that slow down the execution of functions that use the objects.

**Comment by pverrone** [comment]
This bug is actually worse right now. No matter what I do, ALL events are off by an hour. I have tried changing how I handle the local times, just letting the system default, everything. 

Nothing seems to work. What a mess. :(

**Comment by andrzej-aa** [comment]
Hi,
I found the solution, but I tested it only for my timezone, so it needs to be tested for eg. USA and Australia. However, that solution generated another bug - there is no one hour (2nd exactly) when changing time from 2:00 to 3:00. I can't solve it. Maybe someone could find some solution.
Look on my fork. File for testing is named weekcalendar_dst.html.

**Comment by pverrone** [comment]
I was able to manage a workaround for my situation. 

You see, I store everything as Unix timestamp, and, as I mentioned above, I keep all times local. Well, when the UK time changed from BST to GMT, I had a difference of four hours, instead of five (I'm on East Coast USA.)

SO, when creating a new event, it would default to five hours difference, and be off by an hour. (This was consistent with whatever timezone I set my local time to be.) 

I ended up setting a handler in my PHP to change the event time to only four hours so the new events would be consistent. This worked last week, and now works today, after our timechange. The next test will be in March, when UK time is off from ours for two weeks...
